### PR TITLE
fix compiling error of third_party/gmock on Mac

### PR DESCRIPTION
**Pull request by bluealert**
1, fix compiling error of `third_party/gmock` on Mac
2, replace `EXPECT_EQ(false, ...)` with `EXPECT_FALSE(...)`, refer: https://github.com/google/googletest/issues/322\#issuecomment-125645145

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -73,7 +73,7 @@ set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0 -ggdb -fno-omit-frame-po
 
 # ---[ Flags
 if(UNIX OR APPLE)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -Wall -Wextra -Werror -mcx16 -Wno-invalid-offsetof -march=native")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -Wall -Wextra -Werror -Wno-sign-compare -mcx16 -Wno-invalid-offsetof -march=native")
```
Comment (apavlo) [inline CMakeLists.txt]: We don't want to disable this for all linux/unix machines. Only on Mac. So I think you need to add this under the "Darwin" directive right be below it.

**Comment by bluealert** [comment]
@apavlo Professor, I have tested on Ubuntu 16.04(gcc 5.4.0), `-Wno-sign-compare` is also needed on Ubuntu.

**Comment by apavlo** [comment]
Ubuntu 16.04 gcc5 it was working before you added `-Wno-sign-compare` to disable the warning. You are now disabling the warning on **all** platforms, which is not what we want to do. It should **only** be for Darwin.

**Comment by bluealert** [comment]
@apavlo Professor, I found the broken on Ubuntu 16.04 is because the deletion of `SYSTEM` at https://github.com/cmu-db/peloton/blob/master/test/CMakeLists.txt#L36

But, by the [cmake documentation](https://cmake.org/cmake/help/v3.0/command/include_directories.html), it should not use `SYSTEM` there?

**Comment by poojanilangekar** [comment]
@apavlo Are we planning to merge this? 
AFAIK the build currently works for specific versions of Xcode. @rohit-cmu or @AngLi-Leon can you guys confirm?

**Comment by rohit-cmu** [comment]
Works for the following configuration:
Apple LLVM version 8.0.0 (clang-800.0.42.1)
Xcode 8.1
OS X El Capitan Version 10.11.6

**Comment by apavlo** [comment]
@poojanilangekar I don't think we want to add `-Wno-sign-compare` to our make file.

**Comment by poojanilangekar** [comment]
@apavlo So can we close this PR and the corresponding issue? Since we do build on Mac.

**Comment by apavlo** [comment]
@poojanilangekar Yes, if we're already working on the mac then we probably don't want this @bluealert's suggestion about fixing EXPECT_EQ(false...) to EXPECT_FALSE(..) is a good idea. We should do that as part of the clean-up/

**Comment by poojanilangekar** [comment]
@apavlo Added it to the cleanup list.

**Pull request closed without merging**